### PR TITLE
Save Common implementation

### PR DESCRIPTION
**Pull request by emmauss**
This creates a `savecommon` directory for temporary savedata storage, used currently by the NSO NES games. I currently do not know the exact path for save common on console, so any comment about that before this is merged would be good. I also added an optional check for 0 userids, as savecommon uses a 0 userid.

### CODE REVIEW

File: Ryujinx.HLE/HOS/SystemState/UserId.cs [hunk 1]
```diff
@@ -13,11 +13,6 @@ public struct UserId
 
         public UserId(long Low, long High)
         {
-            if ((Low | High) == 0)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/SystemState/UserId.cs]: I don't think those checks should be removed, 0 user ids are NOT valid. In this function it is just used as a special value, but the user id itself doesn't exist and is not used at all when this value is 0.

File: Ryujinx.HLE/HOS/SystemState/UserId.cs [hunk 2]
```diff
@@ -50,11 +45,6 @@ public UserId(string UserIdHex)
                 throw new ArgumentException("Invalid user id!", nameof(UserIdHex));
             }
 
-            if (UserIdHex == "00000000000000000000000000000000")
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/SystemState/UserId.cs]: Same thing here.

**Comment by gdkchan** [comment]
If you're going to remove the checks, maybe its a good idea to also rename the struct to `UInt128` or something like that, since theres nothing special about it anyway and it can be used to store any 128-bits integer.

**Comment by emmauss** [comment]
Renamed the struct.

**Pull request merged**